hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieLogBlock.java

```diff
@@ -121,7 +121,7 @@ public long getLogBlockLength() {
    * new enums at the end.
    */
   public enum HeaderMetadataType {
-    INSTANT_TIME, TARGET_INSTANT_TIME, SCHEMA, COMMAND_BLOCK_TYPE
+    INSTANT_TIME, TARGET_INSTANT_TIME, SCHEMA, COMMAND_BLOCK_TYPE, MIN_INSTANT_TIME, MAX_INSTANT_TIME
```

Comment:
How about we don't add the MIN_INSTANT_TIME & MAX_INSTANT_TIME to the HeaderMetadataType but create an enum like   
```
public enum HeaderMetadataType {
    INSTANT_TIME,
    TARGET_INSTANT_TIME,
    SCHEMA,
    COMMAND_BLOCK_TYPE;
    public enum ArchivedLogHeaderMetadataType {
      MIN_INSTANT_TIME;
     ....
    }
  }
```
It looks like we are just overloading the  `HeaderMetadataType` with information that is pertaining only to archived files which is going to cause confusions when reading actual data blocks. WDYT?

Reply:
I dont have strong opinion either way. These do seem generic enough that we can use for non-archived blocks too.

Also, nested enums could become cumbersome especially since we persist ordinals in metadata. I can test to see if works.

Reply:
It's not about them being just generic, think of it this way - we are going to start dumping MetadataTypes into that one class. Now, different log blocks written in different parts of the code (archival, actual data, may be indexes, may be consolidated metadata) - we will have no idea what header type is used where without looking through the entire code base, having some abstraction here will be very helpful. I agree with your point about ordinals in nested enums, lets test it out and if this doesn't work, we should find another abstraction. let me know if that makes sense.

Reply:
I tried doing this. there are couple problems:
1) java types- HoodieAvroDataBlock relies on header key to be of type: HeaderMetadataType. Nested enum has different java type, so this requires lot more refactoring. enums do not seem to have inheritance support in java, so this likely involves converting enum to string/ordinals and lot of testing to make sure we did not break any backward compatibility

2) enum nesting will end up creating totally new set of ordinals. It can be confusing to see which enum ordinals show. 

tl;dr building better abstraction seems lot more work. Let me know if we want to investigate this further. Likely will have to done as separate task and postpone/abandon merging this until then.